File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int l = flowerbed.size();
        if(l==1){
            if(flowerbed[0] == 1){
                return n>0 ? false:true;
            }else{
                return n>1 ? false:true;
            }
        }
        for(int i = 0;i<l;i++){
            if(flowerbed[i] == 0){
                if(i==0){
                    if(flowerbed[i+1]==0){
                        flowerbed[i] = 1;
                        n--;
                    }
                }else if(i == l-1){
                    if(flowerbed[i-1]==0){
                        flowerbed[i] = 1;
                        n--;
                    }
                }else{
                    if(flowerbed[i+1] == 0 && flowerbed[i-1] == 0){
                        flowerbed[i] = 1;
                        n--;
                    }
                }
            }
            if(n <= 0){
                return true;
            }
        }
        return n>0 ? false:true;
        
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        if(flowerbed.length==1 && flowerbed[0]==0 && n == 1){
            return true;
        }
        int counter = 0;
        if(flowerbed.length>1){
            if(flowerbed[flowerbed.length-1]==0 && flowerbed[flowerbed.length-2]==0){
                counter++;
                flowerbed[flowerbed.length-1] = 1;
            }
            if(flowerbed[0]==0 && flowerbed[1]==0){
                counter++;
                flowerbed[0] = 1;
            }
            
        }
        for(int i = 1;i<flowerbed.length-1;i++){
            if(flowerbed[i-1]==0 && flowerbed[i]==0 && flowerbed[i+1]==0){
                counter++;
                flowerbed[i] = 1;
            }
        }
        
        if(counter>=n) return true;
        return false;
    }
}